### PR TITLE
@wasmer/cli: Replaced oclif with minimist, and fixed flag ordering

### LINKED ISSUE #170: cli flag handling is unintuitive
1. An unhelpful error message (because the number of args isn't checked with `strict = false`):
```
$ wasmer-js run
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be one of type string, Buffer, or URL. Received type undefined
```

2. 🤷‍♂
```sh
$ wasmer-js run --dir=/ test.wasi
```
is interpreted as `wasmer-js run --dir=/ --dir=test.wasi`

### PR DESCRIPTION
**Pull request by torch2424**
closes #170 

This replaces oclif, with [minimist](https://www.npmjs.com/package/minimist), which is a much more bare bones CLI parsing library. From there, I essentially re-implement some of the missing pieces that oclif was providing, but now we have a bit more control on how thngs work / look 😄 

# Screenshots

Screen Shot 2020-01-28 at 5 08 19 PM
Screen Shot 2020-01-28 at 5 09 46 PM
Screen Shot 2020-01-28 at 5 10 15 PM
Screen Shot 2020-01-28 at 5 10 23 PM
Screen Shot 2020-01-28 at 5 10 32 PM
Screen Shot 2020-01-28 at 5 10 41 PM
Screen Shot 2020-01-28 at 5 11 03 PM
Screen Shot 2020-01-28 at 5 11 13 PM
Screen Shot 2020-01-28 at 5 11 36 PM

### CODE REVIEW

**Comment by torch2424** [comment]
@syrusakbary Made requested changes for the CLI to introduce the breaking change to match the functionality of the Wasmer CLI

Screen Shot 2020-01-28 at 5 43 07 PM

File: packages/cli/src/commands/run.ts [hunk 1]
```diff
@@ -1,77 +1,95 @@
...
+        "Options to --mapdir= need to be in the format wasmDir:hostDir"
+      );
+      process.exit(1);
     }
+    preopens[wasm] = host;
+  });
+
+  // Pass non-recognized / remaining args to the wasi module
+  const recognizedArgs = ["--dir", "--mapdir", wasmModuleFileToRun];
+  const wasiArgs = process.argv.slice(3).filter((arg: string) => {
```
Comment (syrusakbary) [inline packages/cli/src/commands/run.ts]: wasiArgs need to have the command (can be the wasm file) as the first argument.

**Comment by torch2424** [comment]
@syrusakbary Made the requested changes 😄

**Comment by torch2424** [comment]
And open SSL works as asked over slack 😄 

Screen Shot 2020-01-28 at 6 03 15 PM

**Pull request merged**